### PR TITLE
Suppress dialog on out of date build

### PR DESCRIPTION
**Pull request by panopticoncentral**
**Customer scenario**

When trying to run an out of date project, a dialog will pop up asking if the user wants to rebuild or run anyway. This dialog has been suppressed for VB/C# in previous versions (I believe C++ uses it) but has not been suppressed for .NET Core projects, leading to annoyance.

**Bugs this fixes:** 

#2358

**Workarounds, if any**

The user could turn off prompting to rebuild before running in the Tools|Options dialog.

**Risk**

Low, this is just turning on a property that the legacy VB/C# project system set.

**Performance impact**

Low, we are just adding a project property.

**Is this a regression from a previous update?**

No

**How was the bug found?**

Ad hoc testing

### CODE REVIEW

**Comment by srivatsn** [comment]
Why not just add the property to managed.designtime.targets? That's imported by common.targets

**Comment by panopticoncentral** [comment]
K, wasn't sure which was the better way to go...

**Comment by panopticoncentral** [comment]
Look OK?

File: src/Targets/Microsoft.Managed.DesignTime.targets [hunk 1]
```diff
@@ -21,6 +21,8 @@
 
     <!--This property sets the default output types supported by the project system.-->
     <SupportedOutputTypes Condition="'$(SupportedOutputTypes)' == ''">Exe;WinExe;Library</SupportedOutputTypes>
+
+    <SuppressOutOfDateMessageOnBuild>true</SuppressOutOfDateMessageOnBuild>
```
Comment (davkean) [inline src/Targets/Microsoft.Managed.DesignTime.targets]: We always check if it's not already set - this enables flavors, SDKs, etc to overrule what we are setting here.

Comment (panopticoncentral) [inline src/Targets/Microsoft.Managed.DesignTime.targets]: This was hardcoded into the legacy project system (couldn't override the behavior), but sure, makes sense...

Comment (davkean) [inline src/Targets/Microsoft.Managed.DesignTime.targets]: It's a IVsHierarchy property (VSHPROPID_SuppressOutOfDateMessageOnBuild) - flavors can override it.

**Comment by panopticoncentral** [comment]
@MattGertz Can you approve?

**Comment by MattGertz** [comment]
Pending director approval.  Behavior is very annoying without this, and inconsistent with other project types.

**Pull request merged**